libnavigation-base/src/main/java/com/mapbox/navigation/base/route/DirectionsSession.kt

```diff
@@ -1,20 +1,24 @@
 package com.mapbox.navigation.base.route
 
 import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.route.model.Route
 
 interface DirectionsSession {
-    var currentRoute: Route?
-
+    fun getRoutes(): List<Route>
     fun setOrigin(point: Point)
     fun getOrigin(): Point
     fun setWaypoints(points: List<Point>)
     fun getWaypoints(): List<Point>
-
-    fun registerRouteObserver(routeObserver: RouteObserver)
-    fun unregisterRouteObserver(routeObserver: RouteObserver)
+    fun setDestination(point: Point)
+    fun getDestination(): Point
+    fun requestRoutes()
```

Comment:
`requestRoutes` was added because
1. without it we cannot make request (now we don't have on-instance create request)
2. if user receive `error/onFailure` he can't re-request route (now we check `setWaypoints`, `setOrigin`, `setDestination` that data is not the same and don't make route request if data is the same)